webapp/src/reducers/updateStateModal/index.ts

```diff
@@ -0,0 +1,26 @@
+import {createSlice} from '@reduxjs/toolkit';
+
+export type UpdateStateModalState = {
+    open: boolean;
+}
+
+const initialState: UpdateStateModalState = {
+    open: false,
+};
+
+export const openUpdateStateModalSlice = createSlice({
+    name: 'openUpdateStateModalSlice',
+    initialState,
+    reducers: {
+        showModal: (state) => {
+            state.open = true;
+        },
+        hideModal: (state) => {
+            state.open = false;
+        },
```

Comment:
```suggestion
        toggleModalVisibility: (state, action) => {
            state.open = action.payload;
        },
```

can pass `true/false` in action

Reply:
Adding a TODO comment.